### PR TITLE
feat(messages): ⚡ improve message list performence

### PR DESCRIPTION
**Pull request by oceanlvr**
### PR Checklist

If you have any questions, you can refer to the [Contributing Guide](https://github.com/emqx/MQTTX/blob/master/.github/CONTRIBUTING.md)

#### What is the current behavior?

a lot of raw `push` operations in the short time lead to lag UI performance, use `vue-virtual-scroller` improve list render performance.

#### Issue Number

None.

#### What is the new behavior?

better performance in the message list component.

#### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

#### Specific Instructions

Are there any specific instructions or things that should be known prior to review?

### CODE REVIEW

**Comment by oceanlvr** [comment]
This commit may improve long list UI performance when the list length is less than `5000` items (with filter way). 
Maybe I should do more code review on the message split part, remove some array copy, etc.

File: src/main.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import VueI18n from 'vue-i18n'
 import VueClipboard from 'vue-clipboard2'
 import path from 'path'
 import log4js from 'log4js'
+import VueVirtualScroller from 'vue-virtual-scroller'
```
Comment (ysfscream) [inline src/main.ts:7]: Perhaps this is the part where we can implement import in the component?

Comment (oceanlvr) [reply src/main.ts:7]: we need this to install `VueVirtualScroller` plugin.

File: src/main.ts [hunk 2]
```diff
@@ -34,6 +35,7 @@ ConnectionInit({ doMigrations: false, undoMigrations: false } as initOptionModel
   Vue.use(VueI18n)
   Vue.use(VueClipboard)
   Vue.use(VueLog4js, config)
+  Vue.use(VueVirtualScroller)
```
Comment (oceanlvr) [inline src/main.ts:38]: we need to import `VueVirtualScroller` in main.ts for Vue plugin install

File: src/main.ts (hunk 1 above)
Comment (oceanlvr) [reply src/main.ts:7]: https://github.com/emqx/MQTTX/pull/685/files/0e36358040c9d89a5d937001dc1a6c7811b39789#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR38

**Review by ysfscream — APPROVED** [review]

**Comment by ysfscream** [comment]
image

The topic tag is missing. It should be a scrollbar width issue.

**Pull request merged**